cmd/pod-scaler/frontend/src/app/Resources/Resources.tsx

```diff
@@ -0,0 +1,140 @@
+import * as React from 'react';
+import {
+  Breadcrumb,
+  BreadcrumbItem,
+  Divider,
+  Drawer,
+  DrawerActions,
+  DrawerCloseButton,
+  DrawerContent,
+  DrawerContentBody,
+  DrawerHead,
+  DrawerPanelContent,
+  Page,
+  PageSection,
+  PageSectionVariants,
+  Text,
+  TextContent,
+  TreeViewDataItem
+} from '@patternfly/react-core';
+import {Link, useHistory, useLocation} from "react-router-dom";
+import {SearchTreeView, toId} from "@app/SearchTreeView/SearchTreeView";
+import {Histograms} from "@app/Histograms/Histograms";
+import {selectNames, selectParents} from "@app/store";
+import {useAppSelector} from "@app/hook";
+import {Location} from "history";
+import {BreadcrumbItemRenderArgs} from "@patternfly/react-core/src/components/Breadcrumb/BreadcrumbItem";
+
+interface ResourcesProps {
+  urlFragment: string;
+  workload: string;
+}
+
+export const Resources: React.FunctionComponent<ResourcesProps> = (props: ResourcesProps) => {
+  const parents = useAppSelector(selectParents);
+  const names = useAppSelector(selectNames);
+  const history = useHistory();
+  const location: Location = useLocation();
+  const [isExpanded, setIsExpanded] = React.useState<boolean>(true);
+  const [activeItem, setActiveItem] = React.useState<TreeViewDataItem>();
+  const drawerRef: React.RefObject<HTMLDivElement> = React.createRef();
+
+  function onOpenClick() {
+    setIsExpanded(true);
+  }
+
+  function onCloseClick() {
+    setIsExpanded(false);
+  }
+
+  React.useEffect(() => {
+    if (activeItem && activeItem.id) {
+      setIsExpanded(false);
+      history.push({
+        pathname: location.pathname,
+        search: "?" + activeItem.id,
+      })
+    }
+  }, [activeItem]) // eslint-disable-line react-hooks/exhaustive-deps
+
+  function onExpand() {
+    if (drawerRef.current) {
+      drawerRef.current.focus();
+    }
```

Comment:
Consider using optional chaining, which is more concise.

```suggestion
    drawerRef.current?.focus();
```

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining